templates/test/dev/custom-builds/patches/custom-builds.yaml

```diff
@@ -8,73 +8,61 @@ spec:
+        set -o pipefail
+        set -o errexit
+
+        curl -L --retry 10 --retry-delay 5 https://github.com/mikefarah/yq/releases/download/v4.6.1/yq_linux_amd64.tar.gz --output /tmp/yq_linux_amd64.tar.gz
+        tar -xzvf /tmp/yq_linux_amd64.tar.gz -C /tmp && mv /tmp/yq_linux_amd64 /usr/bin/yq
+        rm /tmp/yq_linux_amd64.tar.gz
 
-        mkdir -p $HOME/.kube || exit 1
-        cp -i /etc/kubernetes/admin.conf $HOME/.kube/config || exit 1
-        chown $(id -u):$(id -g) $HOME/.kube/config || exit 1
-        # wait for the cluster to come online based on the default kubeadm configuration
-        retrycmd 120 5 10 /usr/bin/kubectl 2>/dev/null cluster-info || exit 1
-        if [ -n "${KUBE_BINARY_URL}" ]; then
-          # stop the kubelet on this node so we can replace it
-          retrycmd 10 5 10 systemctl stop kubelet || exit 1
-          retrycmd_get_tarball 120 5 /tmp/kube-binary.tar.gz ${KUBE_BINARY_URL} || exit 1
-          tar -xzvf /tmp/kube-binary.tar.gz --strip-components=3 -C /usr/bin kubernetes/node/bin/kubelet || exit 1
-          tar -xzvf /tmp/kube-binary.tar.gz --strip-components=3 -C /usr/bin kubernetes/node/bin/kubectl || exit 1
-          chmod +x /usr/bin/kubelet /usr/bin/kubectl || exit 1
-          retrycmd 10 5 10 systemctl start kubelet || exit 1
-          # wait for the cluster to come back online
-          retrycmd 120 5 10 /usr/bin/kubectl 2>/dev/null cluster-info || exit 1
-        fi
-        if [ -n "${KUBE_PROXY_IMAGE_URL}" ]; then
-          retrycmd 10 5 10 kubectl -n kube-system set image daemonset/kube-proxy kube-proxy=${KUBE_PROXY_IMAGE_URL} || exit 1
-        fi
-        if [ -n "${KUBE_APISERVER_IMAGE_URL}" ] || [ -n "${KUBE_CONTROLLER_MANAGER_IMAGE_URL}" ] || [ -n "${KUBE_SCHEDULER_IMAGE_URL}" ]; then
-          retrycmd_get_tarball 120 5 /tmp/yq_linux_amd64.tar.gz https://github.com/mikefarah/yq/releases/download/v4.6.1/yq_linux_amd64.tar.gz || exit 1
-          tar -xzvf /tmp/yq_linux_amd64.tar.gz -C /tmp && mv /tmp/yq_linux_amd64 /usr/bin/yq || exit 1
-        fi
-        if [ -n "${KUBE_APISERVER_IMAGE_URL}" ]; then
-          yq e '.spec.containers[0].image = "${KUBE_APISERVER_IMAGE_URL}"' -i /etc/kubernetes/manifests/kube-apiserver.yaml || exit 1
-        fi
-        if [ -n "${KUBE_CONTROLLER_MANAGER_IMAGE_URL}" ]; then
-          yq e '.spec.containers[0].image = "${KUBE_CONTROLLER_MANAGER_IMAGE_URL}"' -i /etc/kubernetes/manifests/kube-controller-manager.yaml || exit 1
-        fi
-        if [ -n "${KUBE_SCHEDULER_IMAGE_URL}" ]; then
-          yq e '.spec.containers[0].image = "${KUBE_SCHEDULER_IMAGE_URL}"' -i /etc/kubernetes/manifests/kube-scheduler.yaml || exit 1
-        fi
+        export KUBECONFIG=/etc/kubernetes/admin.conf
+        kubectl -n kube-system set image daemonset/kube-proxy kube-proxy="${REGISTRY}/kube-proxy:${IMAGE_TAG}"
+        yq e '.spec.containers[0].image = "${REGISTRY}/kube-apiserver:${IMAGE_TAG}"' -i /etc/kubernetes/manifests/kube-apiserver.yaml
```

Comment:
Do we still need to do this now that you're prepulling the images above?

Reply:
The prepull and retagging is done for the CI artifacts so that kubeadm can pull the correct CI images before we replace them with our images in ACR